c2rust-transpile/src/c_ast/mod.rs

```diff
@@ -152,9 +152,8 @@ impl TypedAstContext {
         }
 
         let mut include_map = vec![];
-        for fileid in 0..files.len() {
+        for (fileid, mut cur) in files.iter().enumerate() {
```

Comment:
It's somewhat wild to me to bind an iteration variable mutably, but sure.

Reply:
It's just like any other `mut` variable, but unlike the vast majority of iteration variables, it gets rebound in the body. The contrast is with a C-style `for` loop where altering the iteration variable in the body would change the iteration structure. It's not wrong, but I was surprised to see this idiom.